strawberryfields/apps/qchem/dynamics.py

```diff
@@ -250,3 +254,84 @@ def prob(samples: list, excited_state: list) -> float:
         raise ValueError("The excited state must not contain negative values")
 
     return samples.count(excited_state) / len(samples)
+
+
+def sample_tmsv(
+    sq: list, t: float, Ul: np.ndarray, w: np.ndarray, n_samples: int, loss: float = 0.0,
+) -> list:
+    r"""Generate samples for simulating vibrational quantum dynamics with a two-mode squeezed input state.
+
+    Given :math:`N` squeezing parameters and :math:`N`-dim normal-to-local transformation matrix,
+    a GBS device with :math:`2N` modes is simulated. The evolution operator acts on only the first
+    :math:`N` modes in the device. Samples are generated by measuring the number of photons in each
+    of the :math:`2N` modes.
+
+    **Example usage:**
+
+    >>> sq = [[0.2, 0.1], [0.8, 0.2]]
+    >>> t = 10.0
+    >>> Ul = np.array([[0.707106781, -0.707106781],
+    >>>                [0.707106781, 0.707106781]])
+    >>> w = np.array([3914.92, 3787.59])
+    >>> n_samples = 5
+    >>> sample_tmsv(sq, t, Ul, w, n_samples)
+    [[0, 0, 0, 0], [0, 0, 0, 0], [0, 1, 0, 1], [0, 1, 0, 1], [0, 2, 0, 2]]
+
+    Args:
+        sq (list[list[float]]): list of two-mode squeezing gate parameters given as [amplitude, phase] for all modes
+        t (float): time in femtoseconds
+        Ul (array): normal-to-local transformation matrix
+        w (array): normal mode frequencies :math:`\omega` in units of :math:`\mbox{cm}^{-1}`
+        n_samples (int): number of samples to be generated
+        loss (float): loss parameter denoting the fraction of lost photons
+
+    Returns:
+        list[list[int]]: a list of samples
+    """
+    if n_samples < 1:
+        raise ValueError("Number of samples must be at least one")
```

Comment:
Same comment as with the coherent sampling PR. I think some of these checks are unnecessary: time and frequencies are positive by definition and the bounds on the loss parameter should show already from `sf.ops.LossChannel()`